pom.xml

```diff
@@ -328,6 +318,11 @@
       <artifactId>reflections</artifactId>
       <version>0.10.2</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-jdk14</artifactId>
+      <version>${slf4j.version}</version>
+    </dependency>
```

Comment:
Adding the SLF4J to JUL bridge, matching Jenkins core. While we do not use SLF4J directly, some of our libraries use it transitively, so this allows all logging to be managed in a single JUL `logging.properties` configuration file.

Reply:
we are still bundling `commons-logging`  via `commons-beansutils` via `jenkins-core` yikes (off to file an issue / fix) ! 

in the interim I think we need to exclude commons-logging to prevent the wrong classes being used (or it being a bit hit and miss which you get depending on something randome or platform dependant), WDYT?

noting that `jcl-over-slf4j` and `log4j-over-slf4j` are currently being pulled in by the jenkins core dependency, but would it be better to manage the versions here to ensure that they are all aligned?

Reply:
> we are still bundling commons-logging via commons-beansutils via jenkins-core yikes (off to file an issue / fix) !

Only via `provided` scope, not in the actual WAR... but yes there is certainly room for improvement in core. Go for it if you want to (note: lots of tricky testing due to the way the `plugin-pom` currently works and its compatibility requirements).

> in the interim I think we need to exclude commons-logging to prevent the wrong classes being used (or it being a bit hit and miss which you get depending on something randome or platform dependant), WDYT?

Sure, and this was logged as a warning (which I didn't notice):

```
[WARNING] Discovered module-info.class. Shading will break its strong encapsulation.
[WARNING] commons-logging-1.2.jar, jcl-over-slf4j-2.0.6.jar define 5 overlapping classes: 
[WARNING]   - org.apache.commons.logging.Log
[WARNING]   - org.apache.commons.logging.LogConfigurationException
[WARNING]   - org.apache.commons.logging.LogFactory
[WARNING]   - org.apache.commons.logging.impl.NoOpLog
[WARNING]   - org.apache.commons.logging.impl.SimpleLog
```

Fixed in commit 1b7dd8d and confirmed the warning is gone

Reply:
> Only via provided scope, not in the actual WAR

yup I forgot those lovely hacks we had. 

> Fixed in commit

👍

Reply:
> off to file an issue / fix

https://github.com/jenkinsci/plugin-pom/issues/676 FTR